test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/persistence/WorkbenchPersistenceIntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+    private static final String PROJECT_NAME = "definition-project";
+    private static final String PROJECT_VERSION = "1.0.0.Final";
+
+    private static final String CONTAINER_ID = "cont-id";
+    private static final String CONTAINER_ALIAS = "cont-alias";
+
+    private WorkbenchClient workbenchClient;
+    private KieServerMgmtControllerClient kieControllerClient;
+
+    @Override
+    protected WorkbenchWithKieServerScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        return deploymentScenarioFactory.getWorkbenchWithKieServerScenarioBuilder().build();
+    }
+
+    @Before
+    public void setUp() {
+        workbenchClient = RestWorkbenchClient.createWorkbenchClient(deploymentScenario.getWorkbenchDeployment().getUrl().toString(),
+                deploymentScenario.getWorkbenchDeployment().getUsername(), deploymentScenario.getWorkbenchDeployment().getPassword());
+        kieControllerClient = new KieServerMgmtControllerClient(deploymentScenario.getWorkbenchDeployment().getUrl().toString() + "/rest/controller",
+                deploymentScenario.getWorkbenchDeployment().getUsername(), deploymentScenario.getWorkbenchDeployment().getPassword());
+    }
+
+    @Test
+    public void testWorkbenchControllerPersistence() {
+        WorkbenchUtils.deployProjectToWorkbench(gitProvider, deploymentScenario.getWorkbenchDeployment(), PROJECT_NAME);
+
+        KieServerInfo serverInfo = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment()).getServerInfo().getResult();
+        String kieServerLocation = serverInfo.getLocation();
+        kieControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, PROJECT_NAME, PROJECT_VERSION, KieContainerStatus.STARTED);
+
+        verifyOneServerTemplateWithContainer(kieServerLocation, CONTAINER_ID);
+
+        scaleToZeroAndToOne(deploymentScenario.getWorkbenchDeployment());
+
+        verifyOneServerTemplateWithContainer(kieServerLocation, CONTAINER_ID);
+    }
+
+    @Test
+    public void testWorkbenchProjectPersistence() {
+        workbenchClient.createOrganizationalUnit(ORGANIZATION_UNIT_NAME, deploymentScenario.getWorkbenchDeployment().getUsername());
+        workbenchClient.createRepository(ORGANIZATION_UNIT_NAME, REPOSITORY_NAME);
+        workbenchClient.createProject(REPOSITORY_NAME, PROJECT_NAME, PROJECT_GROUP_ID, PROJECT_VERSION);
+
+        assertRepositoryAndProjectExists(REPOSITORY_NAME, PROJECT_NAME);
+
+        scaleToZeroAndToOne(deploymentScenario.getWorkbenchDeployment());
+
+        assertRepositoryAndProjectExists(REPOSITORY_NAME, PROJECT_NAME);
+        workbenchClient.deployProject(REPOSITORY_NAME, PROJECT_NAME);
+
+        scaleToZeroAndToOne(deploymentScenario.getWorkbenchDeployment());
+
+        KieServicesClient kieServerClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
```

Comment:
Can be KieServicesClient configured in before method, or is there any reason why can't be?